dragonfly/block/coral_block.go

```diff
@@ -0,0 +1,87 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/block/coral"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"time"
+)
+
+// CoralBlock is a solid block that comes in 5 variants.
+type CoralBlock struct {
+	noNBT
+	solid
+
+	// Type is the type of coral of the block.
+	Type coral.Coral
+	// Dead is whether the coral block is dead.
+	Dead bool
+}
+
+// NeighbourUpdateTick ...
+func (c CoralBlock) NeighbourUpdateTick(pos, changedNeighbour world.BlockPos, w *world.World) {
+	if c.Dead {
+		return
+	}
+	w.ScheduleBlockUpdate(pos, time.Second*5/2)
+}
+
+// ScheduledTick ...
+func (c CoralBlock) ScheduledTick(pos world.BlockPos, w *world.World) {
+	if c.Dead {
+		return
+	}
+
+	adjacentWater := false
+	pos.Neighbours(func(neighbor world.BlockPos) {
+		if _, water := w.Block(neighbor).(Water); water {
```

Comment:
Should use the w.Liquid() method here, as water may also be inside of, say, stairs.